charts/vineyard/templates/master/statefulset.yaml

```diff
@@ -0,0 +1,126 @@
+#  Copyright 2023 The Fluid Authors.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+{{- $replicas := int .Values.master.replicas }}
+{{- $clientPort := int .Values.master.ports.client }}
+{{- $peerPort := int .Values.master.ports.peer }}
+{{- $name := include "vineyard.name" . }}
+{{- $fullName := include "vineyard.fullname" . }}
+{{- $etcdFullname := (printf "%s-%s" $fullName "master") }}
+{{- $chart := include "vineyard.chart" . }}
+{{- $releaseNamespace := .Release.Namespace }}
+{{- $etcdServiceName := (printf "%s-%s" $fullName "master-svc") }}
+
+{{- if not .Values.master.externalEndpoint }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ $fullName }}-master
+  labels:
+    app: {{ $name }}
+    chart: {{ $chart }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    role: vineyard-master
+  {{- if .Values.owner.enabled }}
+  ownerReferences:
+  - apiVersion: {{ .Values.owner.apiVersion }}
+    blockOwnerDeletion: {{ .Values.owner.blockOwnerDeletion }}
+    controller: {{ .Values.owner.controller }}
+    kind: {{ .Values.owner.kind }}
+    name: {{ .Values.owner.name }}
+    uid: {{ .Values.owner.uid }}
+  {{- end }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ $name }}
+      chart: {{ $chart }}
+      release: {{ .Release.Name }}
+      role: vineyard-master
+  serviceName: {{ $fullName }}-master-svc
+  replicas: {{ $replicas }}
+  podManagementPolicy: Parallel
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+      labels:
+        app: {{ $name }}
+        chart: {{ $chart }}
+        release: {{ .Release.Name }}
+        heritage: {{ .Release.Service }}
+        role: vineyard-master
+    spec:
+      {{- if .Values.master.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.master.nodeSelector | trim | indent 8  }}
+      {{- end }}
+      containers:
+        - name: vineyard-master
+          image: {{ .Values.master.image }}:{{ .Values.master.imageTag }}
+          imagePullPolicy: {{ .Values.master.imagePullPolicy }}
+          securityContext:
+            runAsUser: 1001
+            runAsNonRoot: true
+            privileged: false
+          command:
+            - etcd
+            - --name=$(POD_NAME)
```

Comment:
Do we have to give a full url like <pod_name>.<svc_name>.namespace.svc.cluster.local, I'm not sure if users may have customized domain intead of `svc.cluster.local`?

Reply:
Do you mean `--name=<pod_name>.<svc_name>.namespace.svc.cluster.local`? Actually, the name is just used in the internal etcd cluster.

Reply:
> Do you mean `--name=<pod_name>.<svc_name>.namespace.svc.cluster.local`? 

Yes, will`cluster.local` be configured globally in coreDNS config? Refering to https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/#coredns-configmap-options. I'm not sure if it can be modified.